Add Source Code Pro monospace font for code

kapowaz opened the pull request:
This PR adds the [Source Code Pro](https://adobe-fonts.github.io/source-code-pro/) monospace typeface as the default for code samples. I’ve also tweaked the line-height of code which was a little tall by default, as well as some other typography-related metrics.

![img](http://snap.kapowaz.net/d5p3n.png)

In addition to this, I’ve added [Font Face Observer](https://fontfaceobserver.com) to prevent any FOIT whilst we wait for the web fonts to load.

tompave-roo commented:
Personally don't like this kind of ligature-based gliphs.
More objectively, it makes the examples less approachable by people who are not familiar with the trick. It would be better to keep using a font with the same glyphs that are used by 99% of developers.

kapowaz commented:
I’ve updated this PR to use Source Code Pro rather than Hasklig based on feedback from developers. I’ve also added Font Face Observer to let us handle font loading a little better too.

mezis commented on _sass/abstract/_typography.scss:
```diff
@@ -22,29 +57,15 @@
-$title-font: 'Stratos', 'Helvetica Neue', Helvetica, Arial, sans-serif;
-$monospace-font: 'Menlo', 'Ubuntu Mono', 'Lucida Console', 'Monaco', monospace;
+@include font-face-shim('SourceCodePro', 'sourcecodepro/SourceCodePro-Regular', 400);
+@include font-face-shim('SourceCodePro', 'sourcecodepro/SourceCodePro-Semibold', 600);
```
Any particular reason to self-host instead of using e.g. [the Google Fonts CDN](https://fonts.google.com/specimen/Source+Code+Pro)?

kapowaz replied:
Fewer external dependencies? More fine-grained control (should we want it, e.g. custom font subsetting)?

mezis replied:
Makes sense (the dependency part ;) )
